WordLadder/src/assignment3/Main.java

```diff
@@ -125,12 +139,95 @@ public static void initialize() {
 	}
 
 	public static void printLadder(ArrayList<String> ladder) {
-
-		for (int i = 0; i < ladder.size(); i++) {
-			System.out.println(ladder.get(i) + "\n");
+		
+		String start = ladder.get(0);
+		String end = ladder.get(1);
+		int n = ladder.size() - 2;
+		
+		if(ladder.isEmpty()){
+		
+			System.out.println("no word ladder can be found between " 
+			+ start + " and " + end + ".");
+		}
+		
+		else { 
+			
+			System.out.println("a " + n + "-rung word ladder exists between "
+					+ start + " and " + end + ".");
+		
+			for (int i = 0; i < ladder.size(); i++){
+				System.out.println(ladder.get(i) + "\n");
+			}
 		}
-
 	}
 	// TODO
 	// Other private static methods here
+	
+	private static boolean recursDFS(Set<String> dict, Set<String> hs,	
+			ArrayList<String> wordL, String start, String end) {
+
+		boolean flag = false;
+		
+		int count = 0;
+		int i,j,k = 0;
+		int curr = 0;
+		
+		String temp = start;
+		String[] startTemp = start.split("");
+		String[] endTemp = end.split("");
+		
+		if (start.equals(end)){
+			return true;
+		}		
+		
+		for (i = 0; i < end.length(); i++){
+			
+			if ((startTemp[i]).equals(endTemp[i])){
+				curr = i + 1;
+				
+				if (curr >= end.length()){
+					curr = 0;
+					break;
+				}
+			}
+		}
+		
+		while (count != 4){
```

Comment:
just making sure that the index curr is less than the length of startTemp, the first word